codegen/src/main/ops/org/nd4j/codegen/ops/SDBaseOps.kt

```diff
@@ -0,0 +1,1806 @@
+            """.trimIndent()
+        }
+    }
+
+    Op("segmentMean") {
+        javaPackage = "org.nd4j.linalg.api.ops.impl.transforms.custom.segment"
+        Input(NUMERIC, "data") { description = "Data to perform segment max on" }
+        Input(NUMERIC, "segmentIds") { description = "Variable for the segment IDs" }
+        Output(NUMERIC, "output"){ description = "Segment mean output" }
+        Doc(Language.ANY, DocScope.ALL){
+            """
+                Segment mean operation.
+                If data =     [3, 6, 1, 4, 9, 2, 8]
+                segmentIds =  [0, 0, 1, 1, 1, 2, 2]
+                then output = [4.5, 4.666, 5] = [mean(3,6), mean(1,4,9), mean(2,8)]
+                Note that the segment IDs must be sorted from smallest to largest segment.
+                See {@link #unsortedSegmentMean(String, SDVariable, SDVariable, int)} for the same op without this sorted requirement
+            """.trimIndent()
+        }
+    }
+
+    Op("segmentMin") {
+        javaPackage = "org.nd4j.linalg.api.ops.impl.transforms.custom.segment"
+        Input(NUMERIC, "data") { description = "Data to perform segment max on" }
+        Input(NUMERIC, "segmentIds") { description = "Variable for the segment IDs" }
+        Output(NUMERIC, "output"){ description = "Segment min output" }
+        Doc(Language.ANY, DocScope.ALL){
+            """
+                Segment min operation.
+                If data =     [3, 6, 1, 4, 9, 2, 8]
+                segmentIds =  [0, 0, 1, 1, 1, 2, 2]
+                then output = [3, 1, 2] = [min(3,6), min(1,4,9), min(2,8)]
+                Note that the segment IDs must be sorted from smallest to largest segment.
+                See {@link #unsortedSegmentMin(String, SDVariable, SDVariable, int)} for the same op without this sorted requirement
+            """.trimIndent()
+        }
+    }
+
+    Op("segmentProd") {
+        javaPackage = "org.nd4j.linalg.api.ops.impl.transforms.custom.segment"
+        Input(NUMERIC, "data") { description = "Data to perform segment max on" }
+        Input(NUMERIC, "segmentIds") { description = "Variable for the segment IDs" }
+        Output(NUMERIC, "output"){ description = "Segment product output" }
+        Doc(Language.ANY, DocScope.ALL){
+            """
+                Segment product operation.
+                If data =     [3, 6, 1, 4, 9, 2, 8]
+                segmentIds =  [0, 0, 1, 1, 1, 2, 2]
+                then output = [18, 36, 16] = [prod(3,6), prod(1,4,9), prod(2,8)]
+                Note that the segment IDs must be sorted from smallest to largest segment.
+                See {@link #unsortedSegmentProd(String, SDVariable, SDVariable, int)} for the same op without this sorted requirement
+            """.trimIndent()
+        }
+    }
+
+    Op("segmentSum") {
+        javaPackage = "org.nd4j.linalg.api.ops.impl.transforms.custom.segment"
+        Input(NUMERIC, "data") { description = "Data to perform segment max on" }
+ 
```

Comment:
Supports any array type - numeric, boolean, string.
Let's add a DataType.ARRAY or something for this "any" case...

Reply:
As noted earlier, I've added DataType.NDARRAY for this